docs/design/specs/Ecma-335-Augments.md

```diff
@@ -420,3 +420,389 @@ The algorithm is amended as follows:
+
+**Note**: Please note that `DerivedFancyClass` implements the `IFancyTypeName.GetFancyTypeName`
+method via its base class `FancyClass`. While implementing the static interface method in a
+base class is fine, this design proposal doesn't address implementing static interface methods
+in the interfaces themselves or in derived interfaces akin to default interface support.
+
+```
+T Power<T>(T t, uint power)
+    where T : IMultiplication<T>
+{
+    T result = T.One();
+    T powerOfT = t;
+
+    while (power != 0)
+    {
+        if ((power & 1) != 0)
+        {
+            result = T.Multiply(result, powerOfT);
+        }
+        powerOfT = T.Multiply(powerOfT, powerOfT);
+        power >>= 1;
+    }
+
+    return result;
+}
+```
+
+This is an example of polymorphic math where the underlying operators can take arbitrary
+form based on the types involved - you can calculate an integral power of a byte or an int,
+of a float, a double, a complex number, a quaternion or a matrix without much distinction
+with regard to the underlying type, you just need to be able to carry out basic arithmetic
+operations.
+
+```
+T Exponential<T, TFloat>(T exponent) where
+    T : IArithmetic<T>,
+    T : IMultiplicationBy<T, TFloat>
+        
+{
+    T result = T.One();
+    T powerOfValue = exponent;
+    TFloat inverseFactorial = (TFloat)1.0;
+    const int NumberOfTermsInMacLaurinSeries = 6;
+    for (int term = 1; term <= NumberOfTermsInMacLaurinSeries; term++)
+    {
+        result = T.Add(result, (IMultiplicationBy<T, TFloat>)T.Multiply(T.powerOfValue, inverseFactorial));
+        inverseFactorial /= term;
+        powerOfValue = T.Multiply(powerOfValue, exponent);
+    }
+
+    return result;
+}
+```
+
+Another example of polymorphic maths calculating the exponential using the Taylor series,
+usable for calculating the exponential of a matrix.
+
+### II.15.2 Static, Instance and Virtual Methods (page 177)
+
+(Clarify first paragraph)
+
+Static methods are methods that are associated with a type, not with its instances. For
+static virtual methods, the particular method to call is determined via a type lookup
+based on the `constrained.` IL instruction prefix or on generic type constraints but
+the call itself doesn't involve any instance or `this` pointer.
+
+### II.22.26, MethodDef: 0x06
+
+(Edit bulleted section "This contains informative text only" starting at the bottom of page
+233):
+
+Edit section *7.b*: Static | Virtual | !Abstract
+
+(Add new section 41 after the last section 40:)
+
+* 41. If the owner of this method is not an interface, then if Flags.Static is 1 then Flags.Virtual must be 0.
+
+### II.22.27, MethodImpl: 0x19
+
+(Edit bulleted section "This contains informative text only" at the top of the page 237)
+
+Edit section 7: The method indexed by *MethodBody* shall be non-virtual if the method indexed
+by MethodDeclaration is static. Otherwise it shall be virtual.
+
```

Comment:
```suggestion

(Add new section 14 after section 13:)

* 14. If the method indexed by *MethodBody* has the static flag set, the method indexed by *MethodBody* must be indexed via a MethodDef and not a MemberRef. [ERROR]

```

Reply:
This is the change that I believe @AlekseyTs wanted to require that MethodImpl records are MethodDef based.